proto/tripcommon.proto

```diff
@@ -122,3 +133,14 @@ message TurnLane {
   optional uint32 directions_mask = 1;
   optional bool is_active = 2;
 }
+
+enum RoadClass {
```

Comment:
Refactored this enum out of TripLeg and into the top level namespace in https://github.com/valhalla/valhalla/pull/2289/commits/99998bd618e415872f489af8ac200295b14aa4e3 so I could use it in the Location SearchFilter message. Actually much of the diff in this PR is that one refactor commit.

Reply:
Do we have any concerns about backward compatibility here?  This will break any existing users of the protobuf definitions unless they too refactor.

Reply:
I may be wrong but AFAIK the proto definitions are only used within Valhalla, and I'm not aware of a situation where valhalla processes of different versions would be communicating with each other.

Reply:
That was my understanding as well, the external API of Valhalla doesn't include protobuf definitions.

Reply:
correct we have not yet exposed the protobuf API externally so we are ok to do things like thus until we cross that barrier